cmd/auth/auth.go

```diff
@@ -0,0 +1,200 @@
+package auth
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/sha256"
+	_ "embed"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"os/signal"
+	"path"
+	"time"
+
+	"github.com/pkg/browser"
+	"github.com/spf13/cobra"
+)
+
+const (
+	codeChallengeLength = 87
+	supabaseID          = "ibcwmlhcimymasokhgvn"
+	supabasePublicKey   = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoiYW5vbiIsImlhdCI6MTYyOTkzMDc3OCwiZXhwIjoxOTQ1NTA2Nzc4fQ.zcdbd7kDhk7iNSMo8SjsTaXi0wlLNNQcSZkzZ84NUDg"
+	authCallbackAddr    = "localhost:3000"
+	sessionFileName     = "session.json"
```

Comment:
This is probably going to need to be exported so other commands can determine the session filename: maybe drop this into a `pkg/constants.go` file that other commands can utilize?